### PR TITLE
Release 4.0.0 - Management API support + retry policy

### PR DESCRIPTION
**Pull request by stankovski**
* Added support for management APIs
* Added retry policy
* Added automatic retries for throttling and transient errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ This repository contains source code for Azure Notification Hubs .NET SDK as wel
 
 ## Building Code
 
-To build the `Microsoft.Azure.NotificationHubs`, you need support for [.NET Standard 2.0](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).  This requires a minimum of .NET Core 2.0, .NET Framework 4.6.1 or Mono 5.4.  This project ships with two unit test files, one for .NET Core 2.0, and one for .NET Framework 4.6.1 or Mono. 
+To build the `Microsoft.Azure.NotificationHubs`, you need support for [.NET Standard 2.0](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).  This requires a minimum of .NET Core 3.1, .NET Framework 4.6.1 or Mono 5.4.  This project ships with two unit test files, one for .NET Core 3.1, and one for .NET Framework 4.6.1 or Mono.
```
Comment (mpodwysocki) [inline README.md:10]: Should we focus on .NET Standard 2.0 or .NET Standard 2.1?

File: src/Microsoft.Azure.NotificationHubs/Microsoft.Azure.NotificationHubs.csproj [hunk 2]
```diff
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/Microsoft.Azure.NotificationHubs.csproj:4]: Can probably also support .NET Standard 2.1

Comment (stankovski) [reply src/Microsoft.Azure.NotificationHubs/Microsoft.Azure.NotificationHubs.csproj:4]: 2.1 and 2.0 are backwards compatible and we don't have any code that requires 2.1 APIs so I don't see any value in adding 2.1 into the mix. Replacing 2.0 with 2.1 however will break anyone running .NET Core 2.0 apps.

File: README.md (hunk 1 above)
Comment (stankovski) [reply README.md:10]: Let's stick with 2.0

**Pull request merged**